tests/components/whirlpool/test_climate.py

```diff
@@ -0,0 +1,236 @@
+from unittest.mock import AsyncMock, MagicMock, Mock
+
+import pytest
+import whirlpool
+
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_MIDDLE,
+    FAN_OFF,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+)
+from homeassistant.components.whirlpool.climate import AirConEntity, async_setup_entry
+from homeassistant.components.whirlpool.const import DOMAIN
+from homeassistant.const import TEMP_CELSIUS
+
+
+@pytest.fixture(name="aircon_api")
+def aircon_api_fixture():
+    """Set up air conditioner API fixture."""
+    with mock.patch(
+        "homeassistant.components.whirlpool.climate.Aircon"
+    ) as mock_aircon_api:
+        yield mock_aircon_api
+
+
+@pytest.fixture(name="aircon_entity")
+def aircon_entity_fixture(aircon_api):
+    """Set up whirlpool air conditioner entity."""
+    auth = Mock()
+    return AirConEntity("TEST-SAID", auth)
+
+
+async def test_async_setup_entry(hass):
+    """Test the setup of the climate entities."""
+    auth_mock = Mock()
+    auth_mock.get_said_list.return_value = ["SAID1"]
+    async_add_entities_mock = MagicMock()
+    config_mock = MagicMock(
+        data={
+            "entry_id": "0123456789",
+        }
+    )
+    hass.data[DOMAIN] = {config_mock.entry_id: {"auth": auth_mock}}
+
+    await async_setup_entry(hass, config_mock, async_add_entities_mock)
+    async_add_entities_mock.assert_called_once()
+    entities = async_add_entities_mock.call_args.args[0]
+    assert len(entities) == 1
+    assert entities[0].unique_id == "SAID1"
+
+
+async def test_async_setup_entry_no_appliances(hass):
+    """Test the setup of the climate entities when there are no appliances available."""
+    auth_mock = Mock()
+    auth_mock.get_said_list.return_value = []
+    async_add_entities_mock = MagicMock()
+    config_mock = MagicMock(
+        data={
+            "entry_id": "0123456789",
+        }
+    )
+    hass.data[DOMAIN] = {config_mock.entry_id: {"auth": auth_mock}}
+
+    await async_setup_entry(hass, config_mock, async_add_entities_mock)
+    async_add_entities_mock.assert_not_called()
+
+
+async def test_name(aircon_entity, aircon_api):
+    """Test name property."""
+    aircon_api.return_value.connect = AsyncMock()
+    aircon_api.return_value.fetch_name = AsyncMock(return_value="TestZone")
+
+    assert aircon_entity.name == "TEST-SAID"
+    await aircon_entity.async_added_to_hass()
+    assert aircon_entity.name == "TestZone"
+
+
+async def test_unique_id(aircon_entity, aircon_api):
+    """Test name property."""
+    aircon_api.return_value.said = "TEST-SAID-123"
+    assert aircon_entity.unique_id == "TEST-SAID-123"
+
+
+async def test_static_attributes(aircon_entity: AirConEntity, aircon_api):
+    """Test getting attributes."""
+    assert (
```

Comment:
Setup the integration and check the values in the state machine using `hass.states.get`

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Reply:
done